### PR TITLE
Feat: Added Pricing Component

### LINKED ISSUE #295: Pricing interaction
### Description
Create the component as shown in the video attached below.

### Animation Preview
- Screenshot/Video:

https://github.com/user-attachments/assets/50ef404f-2a69-4311-87de-349748366685

> [!NOTE]  
> Please refer to the **Animata contributing guidelines** available [here](https://www.animata.design/docs/contributing) for rules on how to contribute to the project. Let us know in the comments if you’re working on this issue, and feel free to ask any questions!

### Requirements
1. Create a new animated component that matches the provided design.
2. Add example usage of the component in the documentation or storybook.

<details>
  <summary>
    <h3> Guidelines & Best Practices</h3>
  </summary>

- Write reusable and clean code following the project’s style guide.
- The component should be **fully customizable**. The user should be able to modify props such as duration, easing, color, and size where applicable.
- Follow the [Contributing Guidelines](https://www.animata.design/docs/contributing) before submitting your pull request.

</details>

> [!IMPORTANT]  
> To ensure more contributors have the opportunity to participate, we kindly request that:
> - Each contributor submits only **one pull request** related to this issue.
> - If you're interested in working on this issue, please comment below. We will assign the issue on a **first-come, first-served basis**.

---

### Additional Resources
Posted on X by: https://x.com/ln_dev7/status/1821819705977663845

---

> [!Tip]
> Join our Discord community if you need help or have any questions. We're here to help you get started and guide you through the process. [Join now](https://discord.gg/YfvqMf5MTE)

### PR DESCRIPTION
**Pull request by SatyamVyas04**
Added support for a pricing component, that displays plans in a well animated manner, using framer-motion
Here's the demo for the same. Let me know if there are any potential bugs or fallacies in the code 👍🏻 

https://github.com/user-attachments/assets/14aaec23-7894-4993-8fa7-dd73b7f67099

fixes #295

### CODE REVIEW

File: animata/widget/pricing.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+    const animatePrice = () => {
+      const elapsedTime = Date.now() - startTime;
+      const progress = Math.min(elapsedTime / duration, 1);
+      const currentValue = startValue + (endValue - startValue) * progress;
+
+      setPrice(`$${currentValue.toFixed(2)}`);
+
+      if (progress < 1) {
+        requestAnimationFrame(animatePrice);
```
Comment (hari) [inline animata/widget/pricing.tsx:166]: You want to cancel the animation on unmount.

```
let rafId;
...

return () => {
  if (rafId) { 
      cancelAnimationFrame(cancelAnimationFrame);
    }
 }
 ...
 ```

File: content/docs/widget/pricing.mdx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+```jsx file=<rootDir>/animata/widget/pricing.tsx
+
+```
+
+</Steps>
+
+## Credits
+
+Built by [SatyamVyas04](https://github.com/SatyamVyas04)
```
Comment (hari) [inline content/docs/widget/pricing.mdx:38]: Please add: `Inspired by [Nitish Khagwal](https://x.com/nitishkmrk)`

**Review by hari — CHANGES_REQUESTED** [review]
@SatyamVyas04 Looking good. Here are a few things:

1. Please move this to `Sections` category, you will need to create a new one and rename the files as well text.
2. Add little more spacing.

**Comment by SatyamVyas04** [comment]
Working on the animation-unmount now 👍🏻

**Comment by SatyamVyas04** [comment]
This pull request introduces a new `Pricing` component and its associated documentation and stories. The changes include adding the component, creating Storybook stories for it, and updating the documentation to include installation and usage instructions.

### New `Pricing` Component:

* [`animata/section/pricing.tsx`](diffhunk://#diff-b28991ab69c65e4f21a865e3d72fd2448477c128c4f12fffec003ce687fed901R1-R223): Added a new `Pricing` component with customizable plans, width, outer radius, and padding. The component includes interactive elements for selecting billing cycles and plans, with animations for a better user experience.

### Storybook Stories:

* [`animata/section/pricing.stories.tsx`](diffhunk://#diff-e519edaa55797744e9f00e784b88f3b6244d37b423a574dca1707f6c2eb47aedR1-R56): Added Storybook stories for the `Pricing` component, including `Primary` and `CustomPlans` stories to demonstrate different configurations.

### Documentation Updates:

* [`content/docs/section/pricing.mdx`](diffhunk://#diff-1c329ec6c35d5ea841fcb54b4dd0d7e8a4bcfa1bad8acc636e34e255b44c2063R1-R39): Added new documentation for the `Pricing` component, including installation instructions, usage examples, and credits.

### Sidebar Navigation:

* [`config/docs.ts`](diffhunk://#diff-72eed20dacf77aeae0f24473af8500d68560c933a1ee432958b059bac1223650R112-R115): Updated the sidebar navigation to include a new "Section" category with links to the `Pricing` component documentation.

**Review by hari — APPROVED** [review]

**Pull request merged**